### PR TITLE
refactor: Move LocalModCard into own Vue component

### PR DESCRIPTION
**Pull request by GeckoEidechse**
This is in preparation of overhauling the local mods interface.

Basically just moves some stuff from `LocalModsView.vue` to `LocalModCard.vue`.

### CODE REVIEW

File: src-vue/src/views/mods/LocalModsView.vue [hunk 1]
```diff
@@ -5,44 +5,19 @@
...
-            <span v-if="mod.version != null">(v{{ mod.version }})</span>
-            <img
-                v-if="mod.thunderstore_mod_string != null"
-                :title="$t('mods.local.part_of_ts_mod') + '\n' + mod.thunderstore_mod_string"
-                src="/src/assets/thunderstore-icon.png"
-                class="image"
-                height="16"
-            />
-        </el-card>
+    <local-mod-card v-for="mod of mods" v-bind:key="mod.name" :mod="mod" />
```
Comment (Alystrasz) [inline src-vue/src/views/mods/LocalModsView.vue:8]: Is this indented correctly?

File: src-vue/src/components/LocalModCard.vue [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            required: true,
+            type: Object as () => NorthstarMod
+        }
+    },
+    data() {
+        return {
+            global_load_indicator: false,
+        };
+    },
+    computed: {
```
Comment (Alystrasz) [inline src-vue/src/components/LocalModCard.vue:49]: If this property doesn't contain anything, you should remove it.

**Review by Alystrasz — CHANGES_REQUESTED** [review]

File: src-vue/src/views/mods/LocalModsView.vue (hunk 1 above)
Comment (GeckoEidechse) [reply src-vue/src/views/mods/LocalModsView.vue]: Fixed in 15753b6149e1b82c1a08de5a5c7fd13e5299d3f2

File: src-vue/src/components/LocalModCard.vue (hunk 2 above)
Comment (GeckoEidechse) [reply src-vue/src/components/LocalModCard.vue]: Good catch. Removed in c2ae00e5d6c12aca2b637a038416775bc29ae79b ^^

**Review by Alystrasz — APPROVED** [review]
Comments were addressed, code works on Ubuntu 23.04.

**Pull request merged**